packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+  } else if (cancel) {
+    ValoraAnalytics.track(VerificationEvents.verification_cancel)
+    Logger.debug(TAG, 'Verification cancelled')
+  } else if (timeout) {
+    ValoraAnalytics.track(VerificationEvents.verification_timeout)
+    Logger.debug(TAG, 'Verification timed out')
+    yield put(showError(ErrorMessages.VERIFICATION_TIMEOUT))
+    yield put(feelessSetVerificationStatus(VerificationStatus.Failed))
+  }
+  Logger.debug(TAG, 'Done verification')
+
+  yield put(refreshAllBalances())
+}
+
+export function* feelessRestartableVerification(initialWithoutRevealing: boolean) {
+  let isRestarted = false
+  while (true) {
+    const withoutRevealing = !isRestarted && initialWithoutRevealing
+    yield call(navigate, Screens.VerificationLoadingScreen, {
+      withoutRevealing,
+    })
+    yield put(feelessResetVerification())
+    yield call(getConnectedAccount)
+    if (isRestarted || (yield select(isFeelessVerificationStateExpiredSelector))) {
+      yield call(feelessFetchVerificationState)
+    }
+
+    const { verification, restart } = yield race({
+      verification: call(feelessDoVerificationFlow, withoutRevealing),
+      restart: take(Actions.FEELESS_RESEND_ATTESTATIONS),
+    })
+
+    if (restart) {
+      isRestarted = true
+      const { status }: FeelessVerificationState = yield select(feelessVerificationStateSelector)
+      ValoraAnalytics.track(VerificationEvents.verification_resend_messages, {
+        count: status.numAttestationsRemaining,
+      })
+    } else {
+      return verification
+    }
+    Logger.debug(TAG, 'Verification has been restarted')
+  }
+}
+
+export function* feelessDoVerificationFlow(withoutRevealing: boolean = false) {
+  Logger.debug(TAG, '@feelessDoVerificationFlow', 'Starting feeless verification')
+  let receiveMessageTask: Task | undefined
+  let autoRetrievalTask: Task | undefined
+
+  try {
+    const [walletAddress, contractKit, e164Number]: [string, ContractKit, string] = yield all([
+      call(getConnectedUnlockedAccount),
+      call(getContractKit),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.Prepping)),
+    ])
+
+    let feelessVerificationState: FeelessVerificationState = yield select(
+      feelessVerificationStateSelector
+    )
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    // Start by checking again to make sure Komenci is ready. Throws error if not
+    yield call(fetchKomenciReadiness, komenciKit)
+
+    // There should be no instances where the first param is truethy but the second isn't.
+    // Mainly including the second param to satisfy typescript
+    if (
+      !feelessVerificationState.status.isVerified ||
+      !feelessVerificationState.komenci.unverifiedMtwAddress
+    ) {
+      // Start or resume a Komenci session and update state. Throws error if unable to do so
```

Comment:
nit: // Start a new Komenci session if one doesn't exist or we've run out of quota